spec/03-plugins/28-grpc-gateway/01-proxy_spec.lua

```diff
@@ -153,42 +160,1151 @@ for _, strategy in helpers.each_strategy() do
       assert.equal('12', res.headers['grpc-status'])
     end)
 
-    describe("known types transformations", function()
+    describe("grpc <-> json transformations", function()
+      local decimal_positive = 10.5
+      local decimal_negative = -3.75
+      local int32_positive = 11
+      local int64_positive = 2305843009213693951 -- 4611686018427387903 flips the sign?
+      local int32_negative = -15
+      local int64_negative = -2305843009213693951 -- for some reason 2^62 is the maximum negative value
+
+      local H = function( string )
+        return "hello " .. ( string or "" )
+      end
+
+      lazy_setup(function()
+        print ("Updating grpc-gateway plugin (id = " .. plugin1.id .. ") settings: use_proto_names=false; enum_as_name=true; emit_defaults=false")
+
+        assert(bp.plugins:update( { id = plugin1.id }, {
```

Comment:
We should not assume the ordering of the tests, and tests should not depend on each other, otherwise, the test may go flaky.
Thus we should not modify an existing plugin to run tests.

We could add a second route, and attach to it a `grpc-gateway` plugin with different configs. This way we don't need a `lazy_setup` (and a restart) here.

Reply:
Sorry for boilerplate code, I am new to kong unit-tests. What I want to do:

There are three parameters of `protojson` and optimally, the tests should cover all of their combinations. There are 8 combinations of params in total (all of them are booleans).

How to do that?

Reply:
We could write a for loop to iterator over all combinations (like `std::next_permutation` for C++) and put all tests (describe blocks) in the loop body. The other tests should be independent of the setting combinations so they are not conditional, and we also write 3 parts of option-dependent tests, each part checking if the corresponding option works correctly.

Reply:
I understand your suggestion as:
![image](https://user-images.githubusercontent.com/12293198/210978459-22451c65-4c32-45a6-9bb3-a19ee9786bb4.png)
- we have set of unit_tests whose configuration change for every permutation
- we have set of results for Cartesian product of `permutation` x `#unit_tests`
- we have way to map results to test with respect to permutation

What I do not understand how can be plugin settings changed for every permutation. I guess you want to avoid

```
        assert(bp.plugins:update( { id = plugin1.id }, {
          config = {
            proto = "./spec/fixtures/grpc/targetservice.proto",
            use_proto_names = false,
            enum_as_name = true,
            emit_defaults = false,
          },
        } ))
```

Is there another way to change the settings?

Reply:
Maybe we could create `#permutation` x `#unit_tests` of routes (id named with them).

Reply:
something like this? https://github.com/Kong/kong/pull/8297/commits/e51c132857620138e54186d7466f8f6e1585cec3